src/stravalib/client.py

```diff
@@ -1429,10 +1445,13 @@ def get_segment_efforts(
             )
 
         if limit is not None:
+            warn_param_deprecation(limit, "date ranges")
             params["limit"] = limit
 
         result_fetcher = functools.partial(
-            self.protocol.get, "/segments/{segment_id}/all_efforts", **params
+            self.protocol.get,
```

Comment:
This is the actual change here. it's pretty small and shouldn't impact our users at all!